app/src/main/java/com/android/partagix/model/BorrowViewModel.kt

```diff
@@ -0,0 +1,90 @@
+package com.android.partagix.model
+
+import android.location.Location
+import androidx.lifecycle.ViewModel
+import com.android.partagix.model.category.Category
+import com.android.partagix.model.inventory.Inventory
+import com.android.partagix.model.item.Item
+import com.android.partagix.model.loan.Loan
+import com.android.partagix.model.loan.LoanState
+import com.android.partagix.model.user.User
+import com.android.partagix.model.visibility.Visibility
+import java.sql.Date
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+val emptyLoan = Loan("", "", "", "", Date(0), Date(0), "", "", "", "", LoanState.PENDING)
+val emptyItem = Item("", Category("", ""), "", "", Visibility.PUBLIC, 0, Location(""))
+val emptyUser = User("", "", "", "", Inventory("", emptyList()))
+
+class BorrowViewModel(db: Database = Database()) : ViewModel() {
+
+  private val database = db
+
+  private val _loanUiState = MutableStateFlow(emptyLoan)
+  val loanUiState: StateFlow<Loan> = _loanUiState
+
+  private var _itemUiState = MutableStateFlow(emptyItem)
+  val itemUiState: StateFlow<Item> = _itemUiState
+
+  private var _userUiState = MutableStateFlow(emptyUser)
+  val userUiState: StateFlow<User> = _userUiState
+
+  /**
+   * Set the item to borrow and start a new borrow request
+   *
+   * @param item the item to borrow
+   */
+  fun resetBorrow(item: Item, owner: User) {
```

Comment:
I feel like the name of the function doesn't really match, maybe more startBorrow to setToBorrow

Reply:
I had struggle naming this function. `startBorrow` also makes more sense for me.